perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1729,6 +1731,60 @@ public List<RichMember> getRichMembersByIds(PerunSession sess, List<Integer> ids
 		return membersManagerBl.filterOnlyAllowedAttributes(sess, richMembers);
 	}
 
+	@Override
+	public List<FailedCandidate> addMemberCandidates(PerunSession sess, Vo vo, List<MemberCandidate> candidates) {
+		Utils.checkPerunSession(sess);
+
+		List<FailedCandidate> failedCandidates = new ArrayList<>();
+
+		for (MemberCandidate candidate : candidates) {
+			try {
+				if (candidate.getRichUser() != null) {
+					Member member = this.createMember(sess, vo, candidate.getRichUser());
+					getPerunBl().getMembersManagerBl().validateMemberAsync(sess, member);
+				} else if (candidate.getCandidate() != null) {
+					Member member = this.createMember(sess, vo, candidate.getCandidate());
+					getPerunBl().getMembersManagerBl().validateMemberAsync(sess, member);
+				}
+			} catch (Exception e) {
+				failedCandidates.add(new FailedCandidate(candidate, e.getClass().getSimpleName(), e.getMessage()));
+			}
+		}
+
+		return failedCandidates;
+	}
+
+	@Override
+	public List<FailedCandidate> addMemberCandidates(PerunSession sess, Vo vo, List<MemberCandidate> candidates, Group group) throws PrivilegeException {
+		Utils.checkPerunSession(sess);
+
+		// Authorization
+		if (!AuthzResolver.authorizedInternal(sess, "vo-addMemberCandidates_Vo_List<MemberCandidate>_Group_policy", vo) ||
+			!AuthzResolver.authorizedInternal(sess, "group-addMemberCandidates_Vo_List<MemberCandidate>_Group_policy", group)) {
```

Comment:
i think here we only need the group-privilege check (vo object is retrieved from the passed group in the resolver), and in the method above there should be the vo-privilege check. the privilege exception throwing would then be unified for both methods.